### ISSUE TITLE
TypeError: __init__() got an unexpected keyword argument 'custom_head'

### ISSUE THREAD

**Issue by deven-gqc** [timeseriesAI/tsai#597]
My input X and output y both are 3D tensors. 

Using the default `InceptionTimePlus` works as expected.
![image](https://user-images.githubusercontent.com/86305049/197230964-04df36b1-6ea8-4b15-a05d-85dc29516978.png)

However, using any other architecture does not work it seems.
![image](https://user-images.githubusercontent.com/86305049/197231328-e41c0c0d-af7d-4d25-a0c1-9fc9227f31cf.png)

Here is the entire stack trace.
Do other custom architectures not work if `y` is 3-dimensional?

```bash
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
[<ipython-input-21-07879b80fa96>](https://localhost:8080/#) in <module>
      1 learn = TSRegressor(X, y, batch_tfms=batch_tfms, splits=split, 
----> 2                     arch=XceptionTimePlus, metrics=mae, bs=512, train_metrics=True)

5 frames
[/usr/local/lib/python3.7/dist-packages/tsai/tslearner.py](https://localhost:8080/#) in __init__(self, X, y, splits, tfms, inplace, sel_vars, sel_steps, weights, partial_n, train_metrics, bs, batch_size, batch_tfms, shuffle_train, drop_last, num_workers, do_setup, device, arch, arch_config, pretrained, weights_path, exclude_head, cut, init, loss_func, opt_func, lr, metrics, cbs, wd, wd_bn_bias, train_bn, moms, path, model_dir, splitter, verbose)
    117         #                        exclude_head=exclude_head, cut=cut, init=init, arch_config=arch_config)
    118         model = build_ts_model(arch, dls=dls, device=device, verbose=verbose, pretrained=pretrained, weights_path=weights_path,
--> 119                                exclude_head=exclude_head, cut=cut, init=init, arch_config=arch_config)
    120         setattr(model, "__name__", arch.__name__)
    121         try:

[/usr/local/lib/python3.7/dist-packages/tsai/models/utils.py](https://localhost:8080/#) in build_ts_model(arch, c_in, c_out, seq_len, d, dls, device, verbose, pretrained, weights_path, exclude_head, cut, init, arch_config, **kwargs)
    170     else:
    171         pv(f'arch: {arch.__name__}(c_in={c_in} c_out={c_out} device={device}, arch_config={arch_config}, kwargs={kwargs})', verbose)
--> 172         model = arch(c_in, c_out, **arch_config, **kwargs).to(device=device)
    173 
    174     try:

[/usr/local/lib/python3.7/dist-packages/tsai/models/XceptionTimePlus.py](https://localhost:8080/#) in __init__(self, c_in, c_out, nf, nb_filters, coord, norm, concat_pool, adaptive_size, **kwargs)
     75         nf = ifnone(nf, nb_filters)
     76         # Backbone
---> 77         backbone = XceptionBlockPlus(c_in, nf, coord=coord, norm=norm, **kwargs)
     78 
     79         # Head

[/usr/local/lib/python3.7/dist-packages/fastcore/meta.py](https://localhost:8080/#) in __call__(cls, *args, **kwargs)
     38         if type(res)==cls:
     39             if hasattr(res,'__pre_init__'): res.__pre_init__(*args,**kwargs)
---> 40             res.__init__(*args,**kwargs)
     41             if hasattr(res,'__post_init__'): res.__post_init__(*args,**kwargs)
     42         return res

[/usr/local/lib/python3.7/dist-packages/tsai/models/XceptionTimePlus.py](https://localhost:8080/#) in __init__(self, ni, nf, residual, coord, norm, zero_norm, act, act_kwargs, **kwargs)
     59             self.xception.append(XceptionModulePlus(n_in, n_out, coord=coord, 
     60                                                     norm=norm, zero_norm=zero_norm if self.residual and (i-1) % 2 == 0 else False,
---> 61                                                     act=act if self.residual and (i-1) % 2 == 0 else None, **kwargs))
     62         self.add = Add()
     63 

[/usr/local/lib/python3.7/dist-packages/fastcore/meta.py](https://localhost:8080/#) in __call__(cls, *args, **kwargs)
     38         if type(res)==cls:
     39             if hasattr(res,'__pre_init__'): res.__pre_init__(*args,**kwargs)
---> 40             res.__init__(*args,**kwargs)
     41             if hasattr(res,'__post_init__'): res.__post_init__(*args,**kwargs)
     42         return res

TypeError: __init__() got an unexpected keyword argument 'custom_head'
```

**Comment by oguiza** [comment]
Good catch @deven-gqc ,
All models ending in Plus should support the use of a custom_head argument. 
I've added this to 3 models that for some reason were missing it: FCNPlus, MiniRocketPlus and XceptiontimePlus. 
It should work well now. 
I've added an example in the docs to show how to use it. 
It'd be good if you can test it works and close the issue.

**Comment by deven-gqc** [comment]
Thanks for fixing the issue the @oguiza.
I think the commit message `fixes issue #597` did not autoclose the issue.

I think GitHub just wants the commit message to just be `fixes #issue_number` 
I will go ahead and close this! Thanks!
